### DISCUSSION TITLE
自定义provider无法获取useAccess等框架内置插件

### DISCUSSION THREAD

**Discussion by QingShan-Xu** [umijs/umi#12316] [Q&A] [answered]
provider在rootContainer中返回

除修改跟路由外

可以直接调整层级关系吗？
让框架级provider包含用户的provider

**Comment by fz6m** [comment, marked as answer]
用 `innerProvider` 这个，这个位置套的 provider 层级比较低。

通常来说用户没必要套内层的 provider ，除非要使用 antd / 数据流 之类的 hooks ，但不是很推荐这么做，因为通常来说 context 只是配置，配置外置就好了，没必要和上下文强耦合关联起来。

**Reply by QingShan-Xu** [reply]
谢谢你的回答，使用无误。
是的，你说的很对，`context`注入服务，没必要和上下文强耦合关联起来。

原因是项目中使用了全局注册的方式管理模态框，
导致模态框内部取不到框架属性，
合理的方式应是组件注册同时注册相应模态框，但原因种种，还是将错就错吧。
